helix-lsp/src/client.rs

```diff
@@ -981,6 +981,8 @@ impl Client {
                 // Supported(false)
                 _ => return None,
             },
+            // see: https://github.com/microsoft/language-server-protocol/issues/288
+            Some(lsp::TextDocumentSyncCapability::Kind(..)) => false,
             // unsupported
             _ => return None,
```

Comment:
can we remove the blanket case now or turn it into an explicit case. Consedireng that this is pretty subtle (as evident by us getting this wrong) I think explicitly spelling out the other variants would be nicer

Reply:
Ah yeah I think the other variant is just None so that should be fine

Reply:
if the other variant is really just `None` then you could use `.as_ref()?` instead of matching on the option

Reply:
Ah the other variant was not only none it was also when the first match arm was not matched for None for save of TextDocumentSaveOptions, I need to figure out what that should do

Reply:
That's still a good point about the as_ref though

Reply:
refactored it a little bit to use as_ref in two places where None should just return, and then also make sure we are explicitly covering each variant now